pkg/openstack/placement.go

```diff
@@ -0,0 +1,54 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	placementv1 "github.com/openstack-k8s-operators/placement-operator/api/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// ReconcilePlacementAPI -
+func ReconcilePlacementAPI(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+	placementAPI := &placementv1.PlacementAPI{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "placement",
+			Namespace: instance.Namespace,
+		},
+	}
+
+	helper.GetLogger().Info("Reconciling placementAPI", "placementAPI.Namespace", instance.Namespace, "placementAPI.Name", "keystone")
+	op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), placementAPI, func() error {
+		instance.Spec.PlacementTemplate.DeepCopyInto(&placementAPI.Spec)
+		if placementAPI.Spec.Secret == "" {
+			placementAPI.Spec.Secret = instance.Spec.Secret
+		}
+		if placementAPI.Spec.NodeSelector == nil && instance.Spec.NodeSelector != nil {
+			placementAPI.Spec.NodeSelector = instance.Spec.NodeSelector
+		}
+		if placementAPI.Spec.DatabaseInstance == "" {
+			placementAPI.Spec.DatabaseInstance = "openstack" //FIXME: see above
```

Comment:
The FIXME probably need to refer to the comment in https://github.com/openstack-k8s-operators/openstack-operator/blob/becaa3f96db8e694e0a0dbec91a73d2ab89afac0/pkg/openstack/keystone.go#L36

Reply:
I think we can remove it for now